props/prop_base.h

```diff
@@ -379,37 +373,23 @@ class PropBase : CommandParser, Looper, protected SaberBase {
     SaveVolumeIfNeeded();
   }
 
-  // Select preset (font/style)
-  virtual void SetPreset(int preset_num, bool announce) {
-    TRACE(PROP, "start");
+  bool BladeOff() {
 #ifdef IDLE_OFF_TIME
     last_on_time_ = millis();
 #endif
     bool on = SaberBase::IsOn();
     if (on) Off();
-    SaveColorChangeIfNeeded();
-    // First free all styles, then allocate new ones to avoid memory
-    // fragmentation.
+    return on;
+  }
+
+  void UnsetCurrentStyles(int preset_num) {
 #define UNSET_BLADE_STYLE(N) \
     delete current_config->blade##N->UnSetStyle();
     ONCEPERBLADE(UNSET_BLADE_STYLE)
     current_preset_.SetPreset(preset_num);
```

Comment:
This doesn't belong in a function called UnsetCurrentStyles.
Also, in the Update* functions, we probably don't want to call this at all if we can avoid it as it can cause audio glitches.

Reply:
If I recall this is run even during Update* to prevent memory fragmentation when we need to parse/reload the arguments.

Reply:
moved current_preset_.SetPreset(preset_num) to main functions